File: app/src/main/java/com/example/review/New/SaveDataInterface.java
package com.example.review.New;

import java.io.*;

public interface SaveDataInterface {

    byte[] toBytes();

    void toBytes(DataOutputStream dos) throws IOException;

    void loadWith(byte[] rawBytes);

    void loadWith(DataInputStream dis) throws IOException;

    void save(File path);

    void read(File path);
}

File: app/src/main/java/com/example/review/New/SaveDataInterface.kt
package com.example.review.New

import java.io.DataInputStream
import java.io.DataOutputStream
import java.io.File
import java.io.IOException

interface SaveDataInterface {
    fun toBytes(): ByteArray?
    @Throws(IOException::class)
    fun toBytes(dos: DataOutputStream)

    @Throws(IOException::class)
    fun loadWith(rawBytes: ByteArray)

    @Throws(IOException::class)
    fun loadWith(dis: DataInputStream)

    fun save(path: File)
    @Throws(IOException::class)
    fun read(path: File)
}